.github/workflows/build.yml

```diff
@@ -64,75 +64,149 @@ jobs:
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --domain-name pie.design \
+            --region ${{ env.AWS_REGION }} \
+            | jq '
+              [.domainAssociation.subDomains[].subDomainSetting | { prefix: (.prefix // ""), branchName }] |
+              . |= (map(.branchName) | index("pr${{ github.event.number }}")) as $ix |
+              if $ix == null then
+                . + [{branchName: "pr${{ github.event.number }}", prefix: "${{ env.SUB_DOMAIN }}"}]
+              else . end
+            ' -c)
+          echo "DOMAIN_ASSOCIATIONS=$DOMAIN_ASSOCIATIONS" >> $GITHUB_ENV
+      - name: Create Subdomain Association
+        if: github.event_name == 'pull_request' && inputs.deploy
+        shell: bash
+        run: |
+          aws amplify update-domain-association \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --domain-name pie.design \
+            --sub-domain-settings '${{ env.DOMAIN_ASSOCIATIONS }}'
+
+
+      - name: Wait for Amplify deployment
+        if: github.event_name == 'pull_request' && inputs.deploy
+        run: |
+          LATEST_JOB_ID=$(aws amplify list-jobs \
+            --app-id ${{ env.AMPLIFY_ID }} \
+            --branch-name ${{ env.BRANCH_NAME }} \
+            --region ${{ env.AWS_REGION }} --query jobSummaries[0].jobId \
+            --output text)
+          MAX_ATTEMPTS=12
+          SLEEP_SECONDS=10
+
+          attempt_counter=0
+          status=""
+
+          while [ $attempt_counter -lt $MAX_ATTEMPTS ]; do
+            status=$(aws amplify get-job \
+              --app-id ${{ env.AMPLIFY_ID }} \
+              --branch-name ${{ env.BRANCH_NAME }} \
+              --job-id $LATEST_JOB_ID \
+              --query job.summary.status \
+              --output text)
+            echo "Attempt $(( attempt_counter+1 )) of $MAX_ATTEMPTS: Status - $status"
+
+            if [ "$status" == "SUCCEED" ]; then
+              echo "Amplify deployment succeeded."
+              break
+            fi
+
+            if [ "$status" == "FAILED" ]; then
+              echo "Amplify deployment failed."
+              exit 1
+            fi
+
+            let "attempt_counter+=1"
+            sleep $SLEEP_SECONDS
+          done
+
+          if [ "$status" != "SUCCEED" ]; then
+            echo "Amplify deployment did not succeed within the expected time."
+            exit 1
+          fi
 
       # If successful
       - name: Update deployment status (success)
         if: github.event_name == 'pull_request' && inputs.deploy && success()
         uses: chrnorm/deployment-status@v2
         with:
-            token: ${{ secrets.GITHUB_TOKEN }}
-            environment-url: https://pr${{ github.event.number }}.${{ inputs.amplify-app-id }}.amplifyapp.com
-            deployment-id: ${{ steps.deploy.outputs.deployment_id }}
-            state: "success"
+          token: ${{ secrets.GITHUB_TOKEN }}
+          environment-url: https://${{ env.SUB_DOMAIN }}.pie.design/
```

Comment:
Worth pulling this out into a domain variable, or do you think that's best left for the ticket to make it shareable?

Reply:
That's fair. Can you make a note of it for when we make the workflow shareable though?

Reply:
discussed and have added an extra AC to the relevant JIRA